### DISCUSSION TITLE
UnboundLocalError: local variable 'r' referenced before assignment

### DISCUSSION THREAD

**Discussion by lee1046** [semiautomaticgit/SemiAutomaticClassificationPlugin#184] [Q&A] [answered]
I upgraded the version of QGIS on my Mac (Big Sur OSX) and all of sudden Semi-Automatic Classification Plugin stopped working. When trying to create ROI I get the message shown as below. Does anyone have the same issue?
==========================================================================
An error has occurred while executing Python code: 

UnboundLocalError: local variable 'r' referenced before assignment 
Traceback (most recent call last):
  File "/Users/***/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/SemiAutomaticClassificationPlugin/dock/scpdock.py", line 1736, in pointerClickROI
    self.createROI(cfg.pntROI)
  File "/Users/***/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/SemiAutomaticClassificationPlugin/dock/scpdock.py", line 2845, in createROI
    if cfg.np.count_nonzero(r) > 0:
UnboundLocalError: local variable 'r' referenced before assignment

Python version: 3.8.7 (default, May 24 2021, 13:02:34) [Clang 12.0.5 (clang-1205.0.22.9)] 
QGIS version: 3.20.3-Odense Odense, 495fbaecaf 

Python Path:
/Applications/QGIS.app/Contents/MacOS/../Resources/python
/Users/***/Library/Application Support/QGIS/QGIS3/profiles/default/python
/Users/***/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins
/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/statsmodels-0.11.1-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/scipy-1.5.1-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/affine-2.3.0-py3.8.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/patsy-0.5.1-py3.8.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/lib-dynload
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/Fiona-1.8.13.post1-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/rasterio-1.1.5-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/numpy-1.20.1-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/cftime-1.2.1-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/Pillow-7.2.0-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/netCDF4-1.5.4-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/matplotlib-3.3.0-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/GDAL-3.2.3-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS.app/Contents/MacOS/lib/python38.zip
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/snuggs-1.4.7-py3.8.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/geopandas-0.8.1-py3.8.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/pandas-1.1.0-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/Rtree-0.9.4-py3.8.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/opencv_contrib_python-4.3.0.36-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/pyproj-2.6.0-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/click_plugins-1.1.1-py3.8.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages
/Applications/QGIS.app/Contents/MacOS/lib/python3.8
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/numba-0.50.1-py3.8-macosx-10.13.0-x86_64.egg
/Users/***/Library/Application Support/QGIS/QGIS3/profiles/default/python
========================================================================

**Comment by semiautomaticgit** [comment, marked as answer]
Hello @lee1046 , sorry for the late reply.
You can try to follow this guide https://semiautomaticclassificationmanual.readthedocs.io/en/latest/faq.html#i-have-installation-issues-on-mac-os-how-can-i-install-scp-on-mac-os
